finesse/hardware/plugins/temperature/senecak107.py

```diff
@@ -146,6 +150,13 @@ def calc_temp(self, vals: numpy.ndarray) -> numpy.ndarray:
         calc *= self.SCALING_FACTOR
         # Adjusts for minimum temperature limit
         calc += self.MIN_TEMP
+
+        calc[calc > self.MAX_TEMP] = numpy.nan
+        calc[calc < self.MIN_TEMP] = numpy.nan
+
+        if numpy.isnan(calc).any():
+            logging.warn(f"Out-of-range temperature(s) detected: {calc}")
```

Comment:
`logging.warn` seems to be undocumented so I'm assuming it's deprecated.
```suggestion
            logging.warning(f"Out-of-range temperature(s) detected: {calc}")
```

Also, I'm fairly sure calling `logging` to do the logging directly is advised against - we should be using `logging.getLogger()`. However, it seems everywhere else in the code `logging` is called directly. @dalonsoa do you have any insight on this? I know you used `logging.getLogger()` in your `materials` module in PyECN.

Reply:
It is not advised again - [Python documentation uses it with no problem in the first example](https://docs.python.org/3/library/logging.html) - but might be not what you want.

So, `logging.warning`, `logging.error`, etc. all use the `root` logger. If we use `logging.getLogger()`, with no arguments, we also get the `root` logger, therefore `logging.warning("something")` is the same that  `logging.getLogger().warning("something")`.

The difference comes when you have different loggers configured for different parts of the program or want to configure a specific behavior depending on where the logging happen, or what the log is about. In that case you could do `logging.getLogger("gui").warning("something")`, for example, and have a handler configured that send all the logging events sent to the `gui` logger to a pop-up window or something like that.

In summary, in our case, it does not matter. In general, it might.

Reply:
Ah okay, thanks for clarifying!